File: FormalBench-data/diverse/rule_14/DiffEvenOdd.java

import java.io.*;
import java.lang.*;
import java.math.*;

class DiffEvenOdd {

	public static int diffEvenOdd(int[] array) {
		int firstEven = -1;
		int firstOdd = -1;

		for (int num : array) {
			if (!(firstEven == -1 && num % 2 == 0))
				;
			else {
				firstEven = num;
			}
			if (!(firstOdd == -1 && num % 2 != 0))
				;
			else {
				firstOdd = num;
			}
			if (!(firstEven != -1 && firstOdd != -1))
				;
			else {
				break;
			}
		}

		return (firstEven - firstOdd);
	}
}

File: FormalBench-data/FilteredData/translated_c/seed_726_per_category_10_653ade686f/DiffEvenOdd.c
#include "java_arrays.h"
#include <stdint.h>

static int32_t java_sub(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left - (uint32_t)right);
}

static int32_t java_add(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left + (uint32_t)right);
}

int32_t diffEvenOdd(JIntArray array) {
    int32_t firstEven = -1;
    int32_t firstOdd = -1;
    int32_t length = jarray_length(array);

    for (int32_t i = 0; i < length; i = java_add(i, 1)) {
        int32_t num = jarray_get(array, i);
        if (firstEven == -1 && num % 2 == 0) {
            firstEven = num;
        }
        if (firstOdd == -1 && num % 2 != 0) {
            firstOdd = num;
        }
        if (firstEven != -1 && firstOdd != -1) {
            break;
        }
    }
    return java_sub(firstEven, firstOdd);
}
